### PR TITLE
Hardned sha 1

### LINKED ISSUE #49: Hardened SHA-1 support
[Since version 2.13](https://git-scm.com/docs/hash-function-transition/), Git has migrated to a [hardened SHA-1 hash](https://crypto.stackexchange.com/questions/44141/what-is-hardened-sha-1-how-does-it-work-and-how-much-protection-does-it-offer) algorithm.

This algorithm works the same as SHA-1 on most inputs, but will return different hashes for some inputs suspected in exploiting the [SHAttered attack](https://shattered.io/).

Fenrir should use the same algorithm as Git.

### PR DESCRIPTION
**Pull request by Griboedoff**
Closes #49.

### CODE REVIEW

File: Fenrir/UbcCheck.fs [hunk 1]
```diff
@@ -0,0 +1,283 @@
+module Fenrir.UbcCheck
+
+//https://github.com/cr-marcstevens/sha1collisiondetection/blob/master/lib/ubc_check.c
```
Comment (ForNeVeR) [inline Fenrir/UbcCheck.fs:3]: Let's please copy-paste the MIT license header from that file right here.

Comment (Griboedoff) [reply Fenrir/UbcCheck.fs:3]: should i add it to sha1 file too, there are some code from their repo?

Comment (ForNeVeR) [reply Fenrir/UbcCheck.fs:8]: Yep. If we had got our code from somewhere else, we have to follow the code's license restrictions. It's a good thing it's all under MIT.

**Review by ForNeVeR — APPROVED** [review]
Wonderful work, thanks!

**Pull request merged**